packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -156,17 +165,38 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
             return;
         }
 
-        // if the popover handle exists, take it into account
-        let popoverHandleAdjustmentFactor = this.state.isTruncated ? CONTENT_DIV_WIDTH_DELTA : 0;
-        // add a slight bit of buffer space where we don't show the popover, to deal with cases
-        // where everything isn't pixel perfect
-        popoverHandleAdjustmentFactor += .5;
-
-        const isTruncated = this.contentDiv !== undefined &&
-            (this.contentDiv.scrollWidth - popoverHandleAdjustmentFactor > this.contentDiv.clientWidth ||
-            this.contentDiv.scrollHeight > this.contentDiv.clientHeight);
-        if (this.state.isTruncated !== isTruncated) {
-            this.setState({ isTruncated });
+        if (this.contentDiv === undefined) {
+            this.setState({ isTruncated: false });
+            return;
+        }
+
+        const { isTruncated } = this.state;
+
+        // take all measurements at once to avoid excessive DOM reflows.
+        const {
+            clientHeight: containerHeight,
+            clientWidth: containerWidth,
+            scrollHeight: actualContentHeight,
+            scrollWidth: contentWidth,
+        } = this.contentDiv;
+
+        // if the content is truncated, then a popover handle will be present as a
+        // sibling of the content. we don't want to consider that handle when
+        // calculating the width of the actual content, so subtract it.
+        const actualContentWidth = isTruncated
+            ? contentWidth - CONTENT_DIV_WIDTH_DELTA
+            : contentWidth;
+
+        // we of course truncate the content if it doesn't fit in the container. but we
+        // also aggressively truncate if they're the same size with truncation enabled;
+        // this addresses browser-crashing stack-overflow bugs at various zoom levels.
+        // (see: https://github.com/palantir/blueprint/pull/1519)
+        const shouldTruncate = (isTruncated && actualContentWidth === containerWidth)
+            || actualContentWidth > containerWidth
+            || actualContentHeight > containerHeight;
+
+        if (isTruncated !== shouldTruncate) {
```

Comment:
Still need this?

Reply:
@gscshoyru I think we can do away with this check now that this is a pure component.